pyiron_base/jobs/job/runfunction.py

```diff
@@ -782,6 +782,86 @@ def multiprocess_wrapper(
         job_wrap.job.run_static()
 
 
+def generate_calculate_function(write_input_funct=None, collect_output_funct=None):
+    def calculate(input_dict, executable_dict, output_dict={}):
```

Comment:
I'm unhappy about the fact that these args, especially the functions must fulfil certain requirements (`input_dict`, `working_directory` as arguments etc.). I see two options:

- Write all requirements in the doc string (my less favourite option)
- Write where the args of `generate_calculate_function` should come from (I guess `GenericJob._write_input_funct` and `GenericJob._collect_output_funct`?)